src/compiler.ts

```diff
@@ -4917,6 +5124,17 @@ export class Compiler extends DiagnosticEmitter {
       }
       case NodeKind.SUPER: {
         let currentFunction = this.currentFunction;
+        let flow = currentFunction.flow;
+        if (flow.is(FlowFlags.INLINE_CONTEXT)) {
+          let scopedThis = flow.getScopedLocal("this");
+          if (scopedThis) {
+            let base = assert(scopedThis.type.classReference).base;
```

Comment:
Hmm, assert looks very strange in this context, may be should choose better name for that?

Reply:
The idea here is that if there is an explicit `this` type, it must be a class type with a classReference, but the property is nullable for non-class types. What would you suggest?

Reply:
No, I just propose rename method `assert` to smt another because this little bit confuse. May be like `tryGetClassBase`. WDYT?

Reply:
Mh, `assert` is the assertion builtin that makes sure that a value isn't `null`, going away when compiling with `--noAssert`. Can still assign the return value to a var if that makes the code easier to understand.